shared_linux/LinuxSerialDevice/LinuxSerialDevice.cxx

```diff
@@ -0,0 +1,112 @@
+#include "LinuxSerialDevice.h"
+
+// C library headers
+#include <stdio.h>
+#include <string.h>
+
+// Linux headers
+#include <fcntl.h> // Contains file controls like O_RDWR
+#include <errno.h> // Error integer and strerror() function
+#include <termios.h> // Contains POSIX terminal control definitions
+#include <unistd.h> // write(), read(), close()
+#include <sys/ioctl.h> // ioctl()
+#include <sys/time.h>
+
+LinuxSerialDevice::LinuxSerialDevice()
+{
+    // Nothing to do
+}
+
+void LinuxSerialDevice::set_device_file(const char *device_file)
+{
```

Comment:
Can setting the device_file be handled by the constructor?

Reply:
I didn't want it in the constructor because I wanted to declare a global variable for the LinuxSerialDevice and you have to provide the constructor arguments when you declare the global variable, but we don't get this argument until main is running.